added cfpve.cpp

urwifemykids opened the pull request:
Target Branch(es):
335

Issues Addressed:
This update addresses issues related to PVE content when cross-faction configurations are enabled. Specifically, it resolves problems encountered when cross-faction groups participate in dungeons or raids with faction-specific mechanics or creature flags.

Example: In the Icecrown Citadel (ICC) raid, the first player to enter the raid determines the faction alignment for the raid. This can lead to complications such as:

The Gunship Battle becoming problematic, as the enemy faction won't be able to use the cannons.
Mobs that should be friendly becoming enemies and vice versa.
This script ensures that all players have a hidden faction mask that matches the raid leader’s faction. This change prevents faction-related issues during PVE encounters. The script can be extended to handle additional PVE encounters or battlegrounds, which is useful if using NPC bots.

Tests Performed:
The script has been tested and functions as intended.

Shocker committed:
Core/Battlegrounds: Remove temporary currency bonus auras before rewarding player at battleground end

(cherry picked from commit a2f6a2b6ca1473bb2581960132958d869feaa7cd)

Star-lion committed:
Core/Spells: add name for cast_flag_unknown_2 to cast_flag_has_trajectory

(cherry picked from commit 450a6a66c7dc27ea3c0ae1b13436f067adaf5924)

DDuarte committed:
DB/Schema: Change basehp/mana/armor in creature_classlevelstatus to unsigned mediumint

There's no reason to be cheap...

(cherry picked from commit 4d8b06d0bdf691ed81e0ba417ab4d2486702f0dd)

Shauren committed:
Core/Items: Destroy only one item from stack when inserting a gem, not entire stack

(cherry picked from commit 5dc3ccbeab402e8f81cf0b8b38ff6cb252139aa4)

DDuarte committed:
DB/Schema: Increase size of quest_template.RequiredRaces to mediumint

(cherry picked from commit 460f42849fa3eb2eb83d6e580b1c34fc59f9d0f6)

Shauren committed:
Core/Spells: Correct implementation of SPELL_ATTR2_TRIGGERED_CAN_TRIGGER_PROC, SPELL_ATTR3_TRIGGERED_CAN_TRIGGER_PROC_2, SPELL_ATTR3_CAN_PROC_WITH_TRIGGERED and rename them (SPELL_ATTR2_ACTIVE_THREAT, SPELL_ATTR3_NOT_A_PROC, SPELL_ATTR3_CAN_PROC_FROM_PROCS)

(cherry picked from commit b6bc8a50acaf153849f58c8b408e236bb2b70c35)

mdX7 committed:
Core/Vmaps: Fix inconsistency of hitInstance and hitModel to cause wrong area ids (#28632)

Closes #19761
Closes #28326

Co-authored-by: Gosha <284210+Lordron@users.noreply.github.com>
(cherry picked from commit 6ce66659929cbd680a91dd5caa1a5957f30b0716)

Shauren committed:
Build: Perform CMake check_cxx_source_compiles checks using the same C++ standard as used in the project

(cherry picked from commit 34308dd6decd7011ba7ff1196db49b961577c608)

Shauren committed:
Core/Database: Silence deprecation warning on mysql_stmt_bind_param

(cherry picked from commit 5c44ac178798689a4cd31e59bcfeeeecaafa371e)

Shauren committed:
Build: Replaced deprecated exec_program with execute_process

(cherry picked from commit 8eac6224c029c8384f962bb56c768ee5b9c685b9)

Fabi committed:
Tests: Add missing include

(cherry picked from commit 084d9d37b17d7a3b91d26bc224b67daa37c13109)

Shauren committed:
Build: Silence cmake warning about newline in variable

(cherry picked from commit 0e01b72df87d9bff4675df710b8bbd57254beb7e)

Shauren committed:
Build: Remove unneccessary /MP compile option when using ninja

(cherry picked from commit 1083e89da3f8182d377a48d0c46cf2308795f974)

Shauren committed:
Core/Crypto: Rename our Argon2 to avoid possible conflict with library file on some configurations

(cherry picked from commit 4f6b49cccf87638864a636989a41f9ca7eb7a89c)

Shauren committed:
Build: Moved CheckSymbolExists include to correct file

(cherry picked from commit 18fd59d965694b5cd0f62c011875a5d8beaac7e2)

Shauren committed:
Build: Unify setting _BUILD_DIRECTIVE preprocessor macro for all compilers

(cherry picked from commit 078bc6fe3fc764eaa2d528a19fe6d40972ee3537)

Shauren committed:
Build: Normalize executable locations in build directory for all platforms

(cherry picked from commit a7c42f1ed55e586d0d817dd99865aa7c024b0140)

Shauren committed:
Build: Decouple compiler specific settings from os settings

(cherry picked from commit 12d1f4448c30d313e8f0e70d89f29b709f5dd70d)

Shauren committed:
Build: CMake cleanup

* Migrate all add_definitions to target_compile_definitions
* Remove -D from preprocessor definitions added by target_compile_definitions (unneccessary, cmake strips it anyway)
* Fixed NO_BUFFERPOOL not being set on g3d if jemalloc is used
* Moved library/compiler specific compile flag settings spread all over various CMakeLists to their related library/compiler file
* Remove ancient manual link flag settings for worldserver

(cherry picked from commit 77fe2745febab9e4362b09750093d831325e6091)

Shauren committed:
Build: Fixed CMake 3.30 FindBoost deprecation warning

(cherry picked from commit ab5fab5ee9fd0dde509d19ffc26503379a02c41b)

Shauren committed:
Build: Disable maybe-uninitialized warning for GCC that incorrectly triggers with std::optional

(cherry picked from commit d93002f9a96a01258500ca7b5f03fe27e50a7155)

Shauren committed:
Build: Extract hardcoded list of mysql paths to shared cmake variable

Shauren committed:
Build: Replace hardcoded mysql search paths with pattern based approach

Shauren committed:
Core: Remove boost iostreams dependency

(cherry picked from commit dbe8d1f11e844dc73c9ce971421e1d71c41bea3d)

Shauren committed:
Core/Common: Output stdout/stderr from child process without waiting for it to finish

(cherry picked from commit f270686201c84a07f67a9615fd610d917fc8f802)

jackpoz commented:
Note to self: create a branch and change target branch

urwifemykids commented:
Sorry for my ignorance but im not sure what your saying hahaah

jackpoz commented:
Is the original faction ever restored ?

urwifemykids commented:
It doesnt... I forgot about that meh

Aokromes commented:
imho this must be merged into main project.

urwifemykids commented:
#include "ScriptMgr.h"
#include "Player.h"
#include "Group.h"
#include "ObjectAccessor.h"

enum MiscCrossFactionPVE
{
    ZONE_ICECROWN_CITADEL = 4812,
    ICC_MAP_ID = 631,
    ZONE_TRIAL_OF_THE_CHAMPION = 4723,
    TOCHAMPION_MAP_ID = 650,
    ZONE_TRIAL_OF_THE_CRUSADER = 4722,
    TOCRUSADER_MAP_ID = 649,
    ZONE_PIT_OF_SARON = 4813,
    POS_MAP_ID = 658,
    ZONE_HALLS_OF_REFLECTION = 4820,
    HOR_MAP_ID = 668,
    ZONE_FORGE_OF_SOULS = 4809,
    FOS_MAP_ID = 632,
    ZONE_HALLS_OF_STONE = 4264,
    HOS_MAP_ID = 599,
    ZONE_THE_NEXUS = 4265,
    TN_MAP_ID = 576,
    ZONE_WARSONG_GULCH = 3277,
    WSG_MAP_ID = 489,
    ZONE_ARATHI_BASIN = 3358,
    AB_MAP_ID = 529
};

class CfPlayerScript : public PlayerScript
{
public:
    CfPlayerScript() : PlayerScript("CfPlayerScript") {}

    // Called when a player enters the world (logs in or teleports)
    void OnLogin(Player* player, bool firstLogin) override
    {
        HandleFactionChange(player, player->GetMapId());
    }

    // Called when a player changes zones
    void OnUpdateZone(Player* player, uint32 newZone, uint32 /*newArea*/) override
    {
        HandleFactionChange(player, newZone);
    }

private:
    // Store the original faction in a map
    std::unordered_map<uint64, uint32> originalFactionMap;

    void HandleFactionChange(Player* player, uint32 zoneOrMapId)
    {
        static const std::set<uint32> zoneSet = {
            ICC_MAP_ID, TOCHAMPION_MAP_ID, TOCRUSADER_MAP_ID, POS_MAP_ID,
            HOR_MAP_ID, FOS_MAP_ID, HOS_MAP_ID, TN_MAP_ID, WSG_MAP_ID, AB_MAP_ID
        };

        if (zoneSet.count(zoneOrMapId))
        {
            // Change faction to match the group leader
            if (Group* group = player->GetGroup())
            {
                if (Player* leader = ObjectAccessor::FindPlayer(group->GetLeaderGUID()))
                {
                    if (originalFactionMap.find(player->GetGUID()) == originalFactionMap.end())
                    {
                        // Store the original faction
                        originalFactionMap[player->GetGUID()] = player->GetFaction();
                    }
                    player->SetFaction(leader->GetFaction());
                }
            }
        }
        else
        {
            // Restore player's original faction
            auto it = originalFactionMap.find(player->GetGUID());
            if (it != originalFactionMap.end())
            {
                player->SetFaction(it->second);
                originalFactionMap.erase(it); // Clean up the map after restoring
            }
        }
    }
};

void AddSC_cfpve()
{
    new CfPlayerScript();
}

urwifemykids commented:
This is the best I can do for now, if someone wants to help i will accept it, just trying to contribute the best I can, I have added the function to reset faction

acidmanifesto commented on src/server/scripts/Custom/cfpve.cpp:
```diff
@@ -0,0 +1,108 @@
+    CfPlayerScript() : PlayerScript("CfPlayerScript") {}
+
+    // Called when a player enters the world (logs in or teleports)
+    void OnLogin(Player* player, bool firstLogin) override
```
Change bool firstlogin to bool /* firstlogin */

urwifemykids replied:
added

jackpoz commented:
You added the same file twice.

jackpoz commented on cfpve.cpp:
```diff
@@ -0,0 +1,108 @@
+
+private:
+    // Store the original faction in a map
+    std::unordered_map<uint64, uint32> originalFactionMap;
```
This container is not thread-safe and I'm pretty sure OnUpdateZone() can be called by multiple maps at the same time, meaning it will cause race conditions.

Shauren committed:
Core/Common: Catch and log child process creation errors

(cherry picked from commit fff12eafdf971e88ff9b3abe2c6e0a62e050b450)

d3athbl0w commented:
Guys, seeing that it has BG areas such as Warsong Gulch and Arathi Basin, would this be suitable for a Crossfaction Battlegrounds?

Aokromes committed:
DB/Gameobject: Eversong Woods Missing Tattered Chest

closes #30265 by malcrom

ckegg commented:
You can simply just do this to restore to player's original faction without storing it
```player->RestoreFaction();```

You also need to consider pet's faction
```
// set
if (Pet* pet = player->GetPet())
    pet->SetFaction(leader->GetFaction());
// restore
if (Pet* pet = player->GetPet())
    pet->SetFaction(player->GetFaction());
```

If I were you, I will use GetTeamIdInInstance() to play with faction setup instead

Aokromes committed:
DB/Gameobject: The World Tree and the Emerald Dream book update 

closes #30372 by CraftedRO

Aokromes committed:
DB/Creature: Felpaw Village, Felwood Deadwood furbog respawning

closes #30258 by malcrom

Aokromes committed:
DB/Misc: Restore data deleted by mistake

it's bad to push things on late night without triple checking text.

Aokromes committed:
DB/Creature: Jadefire Run, Felwood Respawn Jadefire Tricksters, Betrayers, Hellcallers, and Xavaric

closes #30260 by malcrom

Aokromes committed:
DB/Creature: Remove one Stormwind guard incorrect pathing

closes #30373 by CraftedRO

Shauren committed:
Core/Objects: Eliminate unneccessary temporary objects and double map lookups when building SMSG_UPDATE_OBJECT

(cherry picked from commit 38a209276d9a6b3fb5e657711894a1b696933c59)

Shauren committed:
Core/PacketIO: Skip sending empty SMSG_UPDATE_OBJECT for transports on map changes

(cherry picked from commit f6c37f6791f1a28f5a077f251ec60a541cf0284f)

Aokromes committed:
DB/Creature: Fix incorrect OptionBroadcastTextID for hunter gossip_menu_option

Aokromes committed:
DB/Creature: Fix incorrect OptionBroadcastTextID for hunter gossip_menu_option

missed rows

Aokromes committed:
DB/Quest: City of Light updates

closes #30395 by CraftedRO

Aokromes committed:
DB/Quest: ObjectiveText locale esES and esMX missing text 

closes #30396 by durotar

Aokromes committed:
DB/Condition: Load Into Catapult and Throw Passenger

updates #15220 by sirikfoll

Aokromes committed:
DB/Creature: Icecrown - Ymirheim Updates

closes #30394 by CraftedRO

Aokromes committed:
DB/Trainer: Correct ReqLevel for Darnassus and Thunder Bluff spells

closes  #30403 by CraftedRO

Aokromes committed:
DB/Misc: Fix startup errors from Issue

by malcrom closes #30404

Aokromes committed:
DB/Creature: Add some missing emotes to Birgitte Cranston

closes  #30407 by CraftedRO

Aokromes committed:
DB/Quest: Adjust minimum reputation value required for 'A Special Thank You' (Friendly)

closes #30409 by CraftedRO

Aokromes committed:
DB/Quest: Add few missing queststarters 

closes #30411 by CraftedRO

Aokromes committed:
DB/CreatureTexts: Add proper texts for Rivendark, Furywing, Insidion and Obsidia

closes #30408 by CraftedRO

Aokromes committed:
DB/Creature: Add sniffed waypoints for Hildana Deathstealer for Ymirheim spawn point

closes #30412 by CraftedRO

Aokromes committed:
DB/Gameobject: Culling of Stratholme Love is in the Air spawns

closes #30417 by Foldor1

Aokromes committed:
DB/Gameobject: Borean Tundra missing Brewfest spawns

closes #30423 by Foldor1

Aokromes committed:
DB/Gameobject: Wintergrasp missing Brewfest spawns 

closes #30425 by Foldor1

Aokromes committed:
DB/Gameobject: StormPeaks Missing Brewfest spawns

closes #30426 by Foldor1

Aokromes committed:
typo fix

Aokromes committed:
typo fix 2

Aokromes committed:
DB/Gameobject: Sholazar Basin missing Brewfest spawns

closes #30424 by Foldor1

Aokromes committed:
DB/SAI: Add missing spell to Ravenous Windroc

closes #30429 by CraftedRO

Aokromes committed:
DB/Gameobject: Dragonblight Missing Brewfest spawns 

closes #30430 by Foldor1

Aokromes committed:
DB/Gameobject; Zul'Drak missing Brewfest spawns 

closes #30431 by Foldor1

Aokromes committed:
DB/Gameobject: Grizzly Hills Missing Brewfest spawns

closes #30433 by Foldor1

Aokromes committed:
DB/Gameobject: Howling Fjord: Missing Brewfest spawns

closes #30434 by Foldor1

Aokromes committed:
DB/Gameobject: Shadowmoon Valley Missing Brewfest spawns

closes #30437 by Foldor1

Aokromes committed:
DB/Quest: Update Cloth Scavenging Completion (RewardText)

closes #30438 by CraftedRO

Aokromes committed:
DB/Creature: Update Monte Muzzleshot

updates #15873 by CraftedRO

CraftedRO committed:
Core/Spells: Allow persistent area auras to apply all of their effects upon creation. (#30422)

This fixes the issue where the area auras do not instantly apply their effects.
Actually, they were only applying the last SPELL_EFFECT_PERSISTENT_AREA_AURA;
Flare and Frost Trap have 2 and 3 SPELL_EFFECT_PERSISTENT_AREA_AURA effects respectively, so only the last one would be applied.

Co-Authored-By: zana244 <50482785+zana244@users.noreply.github.com>

Jildor committed:
Scripts/PitOfSaron: Update Ick's reaction state during Explosive Barrage (#30432)

Jildor committed:
Scripts/ForgeOfSouls: Ensure that Bronjahm teleport is executed when entering the second phase (#30428)

Shauren committed:
Core/Spells: Don't treat pets as permanently crowd controlled

Closes #30445

Aokromes committed:
DB/Creature: Update a Firemane Flamecaller spawn position

closes #30448 by CraftedRO

Shauren committed:
Core/vmaps: Improved WMO detection for group models that don't have floor

(cherry picked from commit 28c9474337d0d30bc1d131b12b635c31a98bc97a)

Shauren committed:
Core/vmaps: Removed vmap lookup functions duplicating functionality of each other

(cherry picked from commit 45ee989c70682c001d4467d97bf1ecedcf7dbcc3)

Aokromes committed:
DB/Creature: Aldor Neophyte at Aldor Rise shoud only be draenei

closes #28743 by CraftedRO

Aokromes committed:
DB/Skills: Death Knights shoud start with Journeyman Riding (150)

closes #30449 by CraftedRO

CraftedRO committed:
Core/Unit: Fix DualWield Off-hand hit penalty (#30444)

* Offhand attacks that occur while on-next-hit abilities such as Heroic Strike are queued do not suffer the dual wield to-hit penalty.

Aokromes committed:
DB/Creature: Fix people still on combat after killing tyrannus 

updates #30420  by Odyssey

Commit:
TDB 335.24111 - 2024/11/22

StephanTLavavej committed:
Core/Misc: Remove non-portable hack of including `<__msvc_chrono.hpp>` - will break in future MSVC release (#30460)

(cherry picked from commit 96956b0e5b74884ff8f7bede5c8c11adbb5ee873)

Aokromes committed:
DB/Misc: Fix Alliance GY when dying in Ironforge

closes #30461 by CraftedRO

Shauren committed:
Core/Vmaps: Don't ignore model rotation when calculating liquid level

Closes #30205

(cherry picked from commit 53b4501ce9b072cfffe7de7f348094f443e36950)

heyitsbench committed:
Core/Players: Add separate config option for starting money for Death Knight characters (#30466)

Shauren committed:
Core/Spells: Fixed item level requirements for enchanting spells

Aokromes committed:
DB/Gameobject: Blades Edge Mountains Missing Brewfest spawns

closes #30474 by Foldor1

Aokromes committed:
DB/Gameobject: Zangarmarsh missing Brewfest spawns

closes #30475  by Foldor1

Aokromes committed:
DB/Gameobject: Nagrand  missing Brewfest spawns 

closes  #30476 by Foldor1

CraftedRO committed:
DB/Creature: Update Draenei Youngling PvPFlags

Closes #30479

danielfinke committed:
Scripts/AzuremystIsle: Set correct PvP flags for Draenei Survivor (#30067)

Shauren committed:
DB/Creatures: Add missing PvPFlags to many creatures

Shauren committed:
Dep: Replace basic_deadline_timer with std::chrono based basic_waitable_timer

(cherry picked from commit c81183a6600722f3a9bb4996c0849b530fbdd1b0)

Aokromes committed:
DB/Creature: Buzzard

updates #30497 by CraftedRO

Aokromes committed:
DB/Creature: Update Gates of Ironforge Spirit Healer coordinates

closes #30513 by CraftedRO

Aokromes committed:
DB/Creature: Update two Shattered Sun Marksman SheathState

closes  #30514 by CraftedRO

Aokromes committed:
DB/Gameobject: Culling of Stratholme Hallow's End spawns

closes  #30516 by Foldor1

Aokromes committed:
DB/Gameobject: Culling of Stratholme Feast of Winter Veil spawns

closes  #30515 by Foldor1

Aokromes committed:
DB/Hallow's End: Wickerman Festival updates

closes #30517 by Foldor1

Aokromes committed:
DB/Gameobject: Crystalsong Forest Love is in the Air spawns

closes #30521 by Foldor1

Aokromes committed:
DB/Gameobject: Dragonblight Love is in the Air spawns

closes #30519 by Foldor1

Aokromes committed:
DB/Gameobject: Icecrown Love is in the Air spawns

closes #30520 by Foldor1

Aokromes committed:
DB/Gameobject: The Stockade Love is in the Air spawns

closes #30518 by Foldor1

Aokromes committed:
DB/Gameobject: Borean Tundra Love is in the Air spawns

closes #30522 by Foldor1

Aokromes committed:
DB/Gameobject: Howling Fjord Love is in the Air spawns

closes  #30523 by Foldor1

Aokromes committed:
DB/Gameobject: Blackrock Depths Love is in the Air spawns

closes #30524 by Foldor1

Aokromes committed:
DB/Gameobject: Blackrock Depths Midsummer Fire Festival spawns

closes #30525 by Foldor1

Aokromes committed:
DB/Gameobject: Blackrock Depths Feast of Winter Veil spawns

closes #30526 by Foldor1

Aokromes committed:
DB/Creature: Fix a dberror 

closes #30535 by CraftedRO

Shauren committed:
Core/Time: Fixed WowTime operator<=>

(cherry picked from commit 19aef6882d4ec21f790b67d94a6bd039970156a6)

Shauren committed:
Core/Spells: DynamicObject auras will now use the largest radius found in SPELL_EFFECT_PERSISTENT_AREA_AURA effects of the spell creating them

Shauren committed:
Core/Misc: Reduce code differences between branches

CraftedRO committed:
DB/Quest: Move Priest Garments quest series from c++ to database (#30627)

Co-authored-by: Jonne733 <40151102+Jonne733@users.noreply.github.com>

Shauren committed:
Core/Misc: Reduce code differences between branches

Shauren committed:
Core/Misc: Reduce code differences between branches

Aokromes committed:
DB/Quest: Warlock quest "The Binding"

closes #30665
by Jonne733

Shauren committed:
Core/Misc: Reduce differences between branches and fix data sent in SMSG_AUCTION_COMMAND_RESULT, SMSG_PARTY_MEMBER_STATS, SMSG_BATTLEFIELD_STATUS and implemented missing SMSG_AUCTION_REMOVED_NOTIFICATION

Aokromes committed:
DB/Creature: Fix Witherheart the Stalker equipment 

closes #30671
by CraftedRO

Aokromes committed:
DB/Creature: Add waypoints to Mirelow

closes #30672
by CraftedRO

Shauren committed:
Core/Battlegrounds: Port BattlegroundQueueTypeId changes from master branch

Shauren committed:
Core/PacketIO: Converted BattlegroundPackets to classes

Shauren committed:
Fix build with libstdc++-11

r4d1sh committed:
Core/Spells: Added missing DBC data for some food effects. (#30679)

r4d1sh committed:
Core/Pets: Implemented sound playback when dismissing warlock pets (#30683)

Shauren committed:
Core/Pets: a827230910fd7e3e64bcb5526e2778b8b4752da4 followup reducing differences between branches

Aokromes committed:
DB/Spell: Death Knight T10 Melee 4P Bonus fix 

closes #30451
by CraftedRO

Aokromes committed:
DB/Spell DK blade barrier skill

closes  #27955
by CraftedRO

Aokromes committed:
DB/Quest: Warlock quest series Orb of Orahil

closes #30677
by Jonne733

Aokromes committed:
DB/Quest: Warlock quest "Cleansing of the Orb of Orahil"

closes #30678
by Jonne733

Aokromes committed:
DB/Quest: Warlock quest "The Completed Robe" 

closes #30676
by Jonne733

Aokromes committed:
DB/Creature: Fix model for creature 6546 (Tabetha)

closes  #30687
by CraftedRO

Aokromes committed:
DB/Quest: Update "Lessons Anew" Completion (RewardText)

closes  #30689
by CraftedRO

Aokromes committed:
DB/Quest: Allow "The Multiphase Survey" quest for both sides

closes #30688
by CraftedRO

r4d1sh committed:
Core/Chat: Corrected names and descriptions for GM.LowerSecurity config option and related rbac permission (#30644)

CraftedRO committed:
Core/Battlegrounds: Corrected Arathi Basin near victory sound and score threshold (#30690)

urwifemykids committed:
Merge branch 'TrinityCore:3.3.5' into cfpve

CraftedRO committed:
Core/GameObject: Activated world chests will now despawn after 5 minutes (#30269)

Co-authored-by: Kaytotes <kaytotes@users.noreply.github.com>

Jildor committed:
Core/Groups: align the player's dungeon/raid difficulty with the group's difficulty upon joining, regardless of the player's level. (#30699)

r4d1sh committed:
Core/Spells: Remove ProcEventInfo::GetProcTarget to avoid confusion. (#30435)

Shauren committed:
Scripts/Spells: Fix wrong proc target for spell 23552

Closes #30697

Shauren committed:
Core/Spells: Reduce differences between branches (port refactors from c0602889b57b3a6ab07a7d8ba9b4b91a560f086e)

Aokromes committed:
DB/Creature: Update Knot Thimblejack spawn coordinates

closes  #30700
by CraftedRO

Aokromes committed:
DB/Misc: Remove spawn of Isalien

closes #30698
by Jildor

Shauren committed:
Core/Players: Fixed wrong spell id of Seal of Righteousness being given to players who don't know Judgement of Light

Closes #30642

Shauren committed:
Core/PacketIO: Implemented CMSG_CORPSE_MAP_POSITION_QUERY/SMSG_CORPSE_MAP_POSITION_QUERY_RESPONSE

Shauren committed:
Core/Players: Name some unknown player flags

Shauren committed:
Core/Misc: Reduce differences between branches

Shauren committed:
Core/Transports: Fixed wrong transport guid generator initialization

Shauren committed:
DB: Fix db startup errors related to 48baec04dfd3b654c945df9b4fc2e8feadad760a

Closes #30704

Ovahlord committed:
Core/Scripts: updated more documentation comments for spell script hooks to match with the rest

(cherry picked from commit 71ec8c93acef3dce14eaf52e756ef68a3159050a)

# Conflicts:
#	src/server/game/Spells/SpellScript.h

Aokromes committed:
DB/Quest : Wolves Across the Border The reward text does not take into account the player's gender in the ruRU localization

closes #30713 by Freeman256

Aokromes committed:
DB/Quest : Investigate Echo Ridge The reward text does not take into account the player's gender in the ruRU localization

closes  #30714 by Freeman256

Aokromes committed:
DB/NPC Deputy Willem Missing greeting in the ruRU localization

closes #30715 by Freeman256

Aokromes committed:
DB/Quest : Skirmish at Echo Ridge The reward text does not take into account the player's gender in the ruRU localization

closes #30716 by Freeman256

Aokromes committed:
DB/Text: Correct Marshal McBride's Documents Text in the ruRU localization

closes #30717 by Freeman256

Aokromes committed:
DB/Quest : A Threat Within Text of the task from Cataclysm in the ruRU localization

closes #30712 by Freeman256

Aokromes committed:
DB/Creature: Add missing movement flags to The Lich King

closes #30707
by CraftedRO

Aokromes committed:
DB/Creature: Update Spitelash Serpent Guard

closes #30706
by CraftedRO

Aokromes committed:
Core/Spell: Fix Priest T5 4p Sadist

closes #30702
by r4d1sh

Shauren committed:
Core/Battlegrounds: Fix possible shutdown crash

Closes #30725

Aokromes committed:
DB/QUEST: Plan B, fix gossip

closes #30722 by CraftedRO

Aokromes committed:
DB/Quest: Teldrassil, breadcrumb quests

closes  #30734
by Jonne733

Aokromes committed:
DB/Quest: Teldrassil, missing quest greetings

closes #30735
by Jonne733

Aokromes committed:
DB/Quest: Teldrassil, missing and corrected quest emotes 

closes  #30733
by Jonne733

Aokromes committed:
DB/Creature: Correct respawn time for dwarves in the Lyceum BRD

closes #30732
by CraftedRO

Aokromes committed:
DB/misc: Some corrections in ruRU localization (Part 1)

closes #30723
by Freeman256

Shauren committed:
Core/Objects: Fix orientation for non-unit transport passengers in SMSG_UPDATE_OBJECT and define missing update flag

CraftedRO committed:
Scripts/Scarlet Monastery: Do not start path for npc_scarlet_trainee if combat (#30709)

CraftedRO committed:
Scripts/BlackrockDepths: In the Lyceum both braziers have to be used in order to open both doors. (#30731)

* 1st

* write less, do none

* Update blackrock_depths.cpp

Aokromes committed:
DB/Gameobject: Update position of The Blood Furnace Broggok Cell Door Lever

closes #30745 by  CraftedRO

Aokromes committed:
DB/Creature: Fix Darkspear Dragon Hunter equipment

closes  #30744
by CraftedRO

CraftedRO committed:
Core/Spells: Updates autoshoot spell target in case of client-side target change (#30695)

Shauren committed:
Core/Misc: Reduce differences between branches

Aokromes committed:
DB/Gameobject: Make ICC InstancePortals and Meeting Stone visible for phase 256 too

closes #30757 by CraftedRO

Aokromes committed:
DB/Creature: Update one more Stormwind Guard spawn position

closes #30758 by CraftedRO

Aokromes committed:
DB/Creature: Gorn shoudn't be mounted

closes #30759
by CraftedRO

Aokromes committed:
DB/Creature: Update few Nagrand Spirit Healers spawn position

closes #30767 by CraftedRO

Aokromes committed:
DB/SAI: Add forced despawn in 3 seconds on death to all Ring of Blood npc

close #30764 by CraftedRO

Aokromes committed:
DB/Gossip: Add missing Gurthock gossip_menu

closes #30775
by CraftedRO

Aokromes committed:
DB/Creature: Add proper Ghost aura to all Spirit Healers and remove wrong creature_addon data

closes #30774 by CraftedRO

Aokromes committed:
DB/Quest: Teldrassil, the quest "Moonglade" is missing intro

closes #30779
by Jonne733

Aokromes committed:
DB/Quest: Teldrassil, a few druid quests are missing outro

closes #30780
by Jonne733

Aokromes committed:
DB/Misc: Few more updates to Ring of Blood NPC

closes #30772
by CraftedRO

Aokromes committed:
DB/Quest: Teldrassil, A Few quests are missing Reward next quest

closes #30781
by Jonne733

CraftedRO committed:
Core/Auth: Do not close Authserver at startup when no valid realms found

Aokromes committed:
DB/Creature: Fix Witherbark Scalpers

closes  #30795 by CraftedR

Aokromes committed:
DB/Gameobject: Add two missing quests to a gameobject_queststarter

closes #30793 by CraftedRO

Aokromes committed:
DB/Quest: Teldrassil, fixes to the quest "Sunscorched Shells" outro

closes #30798 by Jonne733

Aokromes committed:
DB/Quest: Teldrassil, the quest "Training the Beast" is missing outro

Aokromes committed:
DB/Quest: Teldrassil, the quest "Vorlus Vilehoof" is missing outro

closes #30800
by Jonne733

Aokromes committed:
DB/Quest: Teldrassil, the quest "Taming the Beast" is missing outro

closes #30801
by Jonne733

Aokromes committed:
DB/SAI: Fix for "Moonglade" quest intro 

closes #30807
by Jonne733

Aokromes committed:
DB/Quest: Two Teldrassil quests are missing PrevQuest

closes #30806
by Jonne733

Aokromes committed:
DB/Quest: Teldrassil, corrected quest texts

closes #30803
by Jonne733

Aokromes committed:
DB/SAI: Denalan, fixes to quest outros

closes #30804
by Jonne733

Aokromes committed:
DB/Gameobject: Fix overlapping Water Barrels/Food Crates in Stormwind City

closes  #27079
by CraftedRO

Aokromes committed:
DB/Gameobject: Update position of the trigger "Pools of Arlithrien Moonwell"

closes #30812
by Jonne733

Aokromes committed:
DB/NPC: Spirit of Sathrah

closes #30811
by Jonne733

Aokromes committed:
DB/SAI: Fix Gahz'rilla random movement on wp2 ended

Closes #30830
updates #24187

Aokromes committed:
DB/Creature: Remove bleed immunity from some npc part 1

closes #30785
by CraftedRO

Aokromes committed:
DB/Creature: Add missing AnimTier Hover for Zidormi

closes #30816
by  CraftedRO

Aokromes committed:
DB/Creature: Update Warmage Kath'leen creature_equip_template

closes #30819
by CraftedRO

Aokromes committed:
DB/SAI: Bogling, add random movement

closes  #30805
by Jonne733

Aokromes committed:
DB/Creature: Westfall Woodworkers are only humans

closes #30825
by CraftedRO

Aokromes committed:
DB/SAI: Fixes for the quest "Mist"

closes #30828
by Jonne733

Aokromes committed:
DB/Creature: Update few more Spirit Healers spawn position

closes #30834
by CraftedRO

Aokromes committed:
DB/SAI: Fix Impsy random movement on wp11 end

closes #30835
by CraftedRO

Aokromes committed:
DB/Creature: Few updates to Hellfire Peninsula

closes #28919
by CraftedRO

Aokromes committed:
DB/Gameobject: Add sniffed spawns and pools to Half-Buried Bottles 

closes  #30837
by CraftedRO

Aokromes committed:
DB/Creature: Add random movement to King Mukla

closes #30838
by CraftedRO

Aokromes committed:
DB/Condition: Teldrassil, Flight Master Vesprystus gossip menu option

closes #30844
by Jonne733

Aokromes committed:
DB/Condition: Teldrassil, Arch Druid Fandral Staghelm gossip menu option

closes #30845
by Jonne733

Aokromes committed:
DB/Quest: Teldrassil, quest is missing quest flag

closes #30849
by Jonne733

Aokromes committed:
DB/Waypiont: Donna and William - Incorrect route points

closes #30719
by CraftedRO

Shauren committed:
Core/Guilds: Fix setting guild notes

Shauren committed:
Dep/Boost: Support boost 1.88

(cherry picked from commit 1b353abbf7deb9dade959e8f5074df5028ef882b)

Aokromes committed:
DB/Creature: Remove Neophyte Guardian from Terokkar Forest - The Bone Wastes

closes #30877
by CraftedRO

Aokromes committed:
DB/Quest/Creature: Blackwind Sabercat 

closes #30865
by CraftedRO

CraftedRO committed:
Scripts/Shadowfang Keep: Check boss state and door state before opening Arugal door (#30891)

Aokromes committed:
DB/Quest: Darkshore, Two quests are missing Reward next quest

closes #30901
by Jonne733

Aokromes committed:
DB/SAI: Thundris Windweaver, fix to quest outro

closes #30900
by Jonne733

Aokromes committed:
DB/NPC: Gubber Blump

closes #30899
by Jonne73

Aokromes committed:
DB/SAI: Fix for "Cave Mushrooms" quest intro

closes #30898
by Jonne733

Aokromes committed:
DB/Quest: Darkshore, missing and corrected quest emotes

closes  #30895
by Jonne733

Aokromes committed:
DB/Quest: Darkshore, corrected quest texts

closes #30896
by Jonne733

chaosua committed:
DB/Creature: Update ruRU locale for shaman totems translation (#30903)

Aokromes committed:
DB/Gameobject: Remove Courtyard Door Lever script in SFK to prevent people being locked inside 

closes #30907
by CraftedRO

CraftedRO committed:
Scripts/Dire Maul: Allow players to open The Prince's Chest if Prince Tortheldrin is killed (#30866)

offl committed:
Scripts/Spells: Implement Putrid Mushroom Primer spell (#30914)

CraftedRO committed:
Core/Misc: Minor corrections to urls and grammar in logs and comments (#30917)

offl committed:
Scripts/Spells: Get rid of some database spell scripts (#30915)

offl committed:
Scripts/Spells: Implement generic spell script to cancel aura by id from spell effect basepoints (#30908)

offl committed:
Scripts/EscapeFromDurnholdeKeep: Modernize scripts

offl committed:
Scripts/TheSlavePens: Modernize scripts

mdX7 committed:
Core/Spells: Reserved custom spell attribute SPELL_ATTR0_CU_CAN_TARGET_ANY_PRIVATE_OBJECT

offl committed:
Scripts/Scholomance: Modernize scripts (#30920)

offl committed:
Scripts/TheBlackMorass: Modernize scripts

offl committed:
Scripts/SteamVault: Modernize scripts (#30948)

offl committed:
Scripts/TheArcatraz: Modernize scripts (#30945)

offl committed:
Scripts/TheBotanica: Modernize Thorngrin script (#30943)

offl committed:
Scripts/SethekkHalls: Modernize scripts (#30935)

offl committed:
Scripts/Mechanar: Modernize scripts

offl committed:
Scripts/AuchenaiCrypts: Modernize Shirrak script (#30938)

Aokromes committed:
DB/Quest: Darkshore, fix outro of quest "How Big a Threat?"

closes #30911
by Jonne733

Aokromes committed:
DB/Quest: Darkshore, misc fixes to the quest "Therylune's Escape"

closes #30912
by Jonne733

Aokromes committed:
DB/Loot: Finish updating vanilla skinning loot

closes #30904
by offl

Aokromes committed:
DB/Loot: Add chance to TBC skinning loot 

closes #30922
by offl

CraftedRO committed:
Core/Spells: Fix hunter pets not losing happiness when dismissed (#30756)

CraftedRO committed:
Core/Spells: Properly retrieve pet to dismiss for spells with SPELL_ATTR1_DISMISS_PET (#30928)

offl committed:
Scripts/AuchenaiCrypts: Small corrections to scripts (#30959)

CraftedRO committed:
Core/Creatures: Update creature GenerateHealth rounding formula (#30971)

CraftedRO committed:
Core/Misc: Don't teleport players to homebind location when logging out inside instance without a group (#30982)

Shauren committed:
Core/Players: Fix Titan's Grip weapon type restrictions

Closes #20972
Closes #30747

CraftedRO committed:
Core/Pets: Allow pet commands while hunter is under SPELL_AURA_MOD_PACIFY aura

Shauren committed:
Scripts/The Escape From Durnholde: Fixed Thrall not resetting state after combat

Closes #28797
Closes #30739

Shauren committed:
Core/Players: Unbreak dual wield

urwifemykids committed:
Merge branch 'TrinityCore:3.3.5' into cfpve
